packages/@uppy/drop-target/src/index.js

```diff
@@ -0,0 +1,133 @@
+const { Plugin } = require('@uppy/core')
+const getDroppedFiles = require('@uppy/utils/lib/getDroppedFiles')
+const toArray = require('@uppy/utils/lib/toArray')
+
+/**
+ * Drop Target plugin
+ *
+ */
+module.exports = class DropTarget extends Plugin {
+  static VERSION = require('../package.json').version
+
+  constructor (uppy, opts) {
+    super(uppy, opts)
+    this.type = 'acquirer'
+    this.id = this.opts.id || 'DropTarget'
+    this.title = 'Drop Target'
+
+    // Default options
+    const defaultOpts = {
+      target: null,
+    }
+
+    // Merge default options with the ones set by user
+    this.opts = { ...defaultOpts, ...opts }
+    this.removeDragOverClassTimeout = null
+  }
+
+  addFiles = (files) => {
+    const descriptors = files.map((file) => ({
+      source: this.id,
+      name: file.name,
+      type: file.type,
+      data: file,
+      meta: {
+        // path of the file relative to the ancestor directory the user selected.
+        // e.g. 'docs/Old Prague/airbnb.pdf'
+        relativePath: file.relativePath || null,
+      },
+    }))
+
+    try {
+      this.uppy.addFiles(descriptors)
+    } catch (err) {
+      this.uppy.log(err)
+    }
+  }
+
+  handleDrop = (event) => {
+    event.preventDefault()
+    event.stopPropagation()
+    clearTimeout(this.removeDragOverClassTimeout)
+
+    // 2. Remove dragover class
```

Comment:
The numbering is a little funny in the comments here in `handleDrop()` and below in `handleDragOver`

Reply:
right, because I think that’s the order of events: 1 is happening in drag over and 2-3 later on in drop